scripts/query-tracing.bcc

```diff
@@ -0,0 +1,260 @@
+#! /usr/bin/python3
+# @lint-avoid-python-3-compatibility-imports
+#
+# query_tracing - Show VAST query statistics
+
+from __future__ import print_function
+from bcc import BPF, USDT
+import ctypes
+from ctypes import c_int, c_ulonglong, c_byte, c_ushort
+from sys import argv
+
+
+def usage():
+    print("USAGE: %s /path/to/libvast.so" % argv[0])
+    exit()
+
+
+# arguments
+if len(argv) != 2:
+    usage()
+
+vast_path = argv[1]
+
+u = USDT(path=vast_path)
+# TODO: Later versions of bcc allow writing 'vast:streaming_importer' to explicitly specify
+#       the provider, but the one from ubuntu doesn't have that feature yet.
+u.enable_probe("query_new", "query_new_handler")
+u.enable_probe("query_meta_index", "query_meta_index_handler")
+u.enable_probe("query_resume", "query_resume_handler")
+u.enable_probe("query_partition_done", "query_partition_done_handler")
+u.enable_probe("query_partition_error", "query_partition_error_handler")
+u.enable_probe("query_done", "query_done_handler")
+
+# load BPF program
+b = BPF(
+    text="""
+#include <uapi/linux/ptrace.h>
+
+enum event_types {
+    EVENT_QUERY_STATE = 0,
```

Comment:
Should this be named `EVENT_QUERY_DONE`?

Reply:
Good point.